src/main/java/org/ow2/proactive/catalog/service/CatalogObjectService.java

```diff
@@ -125,11 +130,7 @@ public CatalogObjectMetadata createCatalogObject(String bucketName, String name,
             if (catalogObject == null) {
                 String contentTypeOfFile = APPLICATION_OCTET_STREAM;
                 try {
-                    InputStream is = new BufferedInputStream(new ByteArrayInputStream(file.getContent()));
-                    String guessedContentTypeOfFile = URLConnection.guessContentTypeFromStream(is);
-                    if (guessedContentTypeOfFile != null) {
-                        contentTypeOfFile = guessedContentTypeOfFile;
-                    }
+                    contentTypeOfFile = getFileMimeType(file);
```

Comment:
Can the try/catch be moved in the `getFileMimeType` method?
Also, the assignment `contentTypeOfFile = APPLICATION_OCTET_STREAM;` could be done only in the catch clause.